internal/controllers/gateway/dataplane_client.go

```diff
@@ -0,0 +1,25 @@
+package gateway
+
+import (
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	k8sobj "github.com/kong/kubernetes-ingress-controller/v2/internal/util/kubernetes/object"
+)
+
+// DataPlane is a common interface that is used by reconcilers to interact
+// with the dataplane.
+//
+// TODO: This can probably be used in other reconcilers as well.
```

Comment:
Do you mind creating an issue for this? Or make them use it straight away in this PR (I believe that shouldn't be many lines to change given it'd be just swapping the type in constructors)?

Reply:
Done.